NetSDK/Services/Cache/Classes/SegmentCache.cs

```diff
@@ -10,20 +11,31 @@ namespace Splitio.Services.Cache.Classes
 {
     public class SegmentCache : ISegmentCache
     {
-        private ConcurrentDictionary<string, SelfRefreshingSegment> segments;
+        private ConcurrentDictionary<string, Segment> segments;
 
-        public void AddToSegment(string segmentName, HashSet<string> segmentKeys)
+        public SegmentCache(ConcurrentDictionary<string, Segment> segments)
```

Comment:
The dictionary is an implementation detail. Should not be present in the constructor.